android/persistkit/src/main/java/co/omise/persistkit/Crypter.kt

```diff
@@ -61,14 +60,15 @@ open class Crypter(private val aliasKeyName: String) {
         keyStore.deleteEntry(aliasKeyName)
     }
 
-    open fun encrypt(plainData: ByteArray): ByteArray {
+    fun encrypt(plainData: ByteArray): ByteArray {
+        val cipher = Cipher.getInstance(TRANSFORMATION_SYMMETRIC)
         cipher.init(Cipher.ENCRYPT_MODE, key, parameterSpec)
-        return cipher.doFinal(plainData)
+        return Base64.encode(cipher.doFinal(plainData), Base64.DEFAULT)
```

Comment:
@pitiphong-p why do we need to determine data format?  🤔 also we encode base64 after encrypted content if we want to store non-binary data why shouldn't we encode it on basic database too?.

Reply:
My initial assumption was that there is a null byte in the encoded data which can cause the exception to be thrown. So we tried to avoid this by encode the data into Base 64 data instead.

Do you have any idea on this? How could we improve this?

Reply:
@pitiphong-p I wonder what exception was thrown? from code that fixed I guess that the problem possibly comes from storing content to database but sql blob possible to store image (which likely to contain a lot of null bytes) and tried to reproduce by putting null byte to basic database everything seem to be work fine. so I have no idea for now why 😅

Reply:
I can't recall what exception exactly, I believe it's something about bad padding or invalid block size

Reply:
that's pretty weird. from my understand aes/gcm/nopadding mean we don't have either block (since it's stream cipher) or padding since we use no-padding 🤔

Reply:
Yeah, I still have no idea what caused this. Do you have any comment or recommendations?

Reply:
I tried the commit 9efa671 which stops sharing the cipher instance and it works now. We may need to retest again with this commit.